hostresolver/hostresolver.go

```diff
@@ -0,0 +1,82 @@
+package hostresolver
+
+import (
+	"net"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/miekg/dns"
+	"github.com/patrickmn/go-cache"
+)
+
+// HostResolver used for host resolver
+type HostResolver struct {
+	Enabled      bool
+	Cache        *cache.Cache
+	ResolvConfig string
+	ResolvDepth  int
+}
+
+// NewHostResolver init host resolver
+func NewHostResolver(enabled bool, resConfig string, resDepth int) *HostResolver {
```

Comment:
In the way to be homogeneouse ith the rest of the code, can you delete the `NewHostResolver` and initialize the structure directly please?